### PR TITLE
feat(QueryService): Handle batching for queries that do not return total matched

### PR DESCRIPTION
**Pull request by adrianossi**

### CODE REVIEW

File: src/services/QueryService.ts [hunk 1]
```diff
@@ -116,7 +116,7 @@ export class QueryService<T> {
   }
 
   private async recursiveQueryPull({ count = 0, start = 0 }) {
-    const [nextStart, nextCount] = this.getNext(start, count);
+    const { nextStart, nextCount } = this.getNext(start, count);
```
Comment (bvkimball) [inline src/services/QueryService.ts:119]: keep array syntax

File: src/services/QueryService.ts [hunk 2]
```diff
@@ -126,15 +126,18 @@ export class QueryService<T> {
...
+    }
     return (nextStart < total && count !== 0) ? nextCount : 0;
   }
 
   private getNext(start: number, count: number) {
     const nextStart = start + count;
     const alreadyFetched = nextStart - this.parameters.start;
     const nextCount = this.parameters.count - alreadyFetched;
-    return [nextStart, nextCount];
+    return { nextStart, nextCount, alreadyFetched };
```
[the remark this answers is not in the document]
Comment (bvkimball) [reply src/services/QueryService.ts:140]: I think alreadyFetched is unessesary to return

File: src/services/QueryService.ts [hunk 3]
```diff
@@ -126,15 +126,18 @@ export class QueryService<T> {
   }
 
   private shouldPullMoreRecords({ count = 0, start = 0, total = 0 }) {
-    const [nextStart, nextCount] = this.getNext(start, count);
+    const { nextStart, nextCount, alreadyFetched } = this.getNext(start, count);
+    if (!this.parameters.showTotalMatched) {
+      return (alreadyFetched < this.parameters.count && count !== 0) ? nextCount : 0;
```
Comment (bvkimball) [inline src/services/QueryService.ts:131]:
```
return (count > 0) ? nextCount : 0;
```

**Review by bvkimball — DISMISSED** [review]

File: src/services/QueryService.ts (hunk 3 above)
Comment (adrianossi) [reply src/services/QueryService.ts:131]: I was using alreadyFetched to short circuit when we know we have already gotten as many as the user requested, so we don't have to make a call that returns 0 records every time

File: src/services/QueryService.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (adrianossi) [reply src/services/QueryService.ts:140]: I was using alreadyFetched to short circuit when we know we have already gotten as many as the user requested, so we don't have to make a call that returns 0 records every time

File: src/services/QueryService.ts (hunk 3 above)
Comment (bvkimball) [reply src/services/QueryService.ts:131]: But you don't need to because you have next count

**Review by hiqbal01 — DISMISSED** [review]

**Review by hiqbal01 — APPROVED** [review]

**Pull request merged**